src/server/list_family.cc

```diff
@@ -1226,8 +1226,8 @@ void ListFamily::BPopGeneric(ListDir dir, CmdArgList args, ConnectionContext* cn
     case OpStatus::TIMED_OUT:
       return rb->SendNullArray();
     case OpStatus::KEY_MOVED:
-      // TODO: proper error for moved
-      return cntx->SendError("-MOVED");
+      // if key is moved the cluster is definitely exist and we must get an error
+      return cntx->SendError(*cluster::SlotOwnershipErrorStr(*transaction->GetUniqueSlotId()));
```

Comment:
```suggestion
      auto error = cluster::SlotOwnershipErrorStr(*transaction->GetUniqueSlotId());
      CHECK(error.has_value());
      return cntx->SendError(*error);
```

Reply:
We already know that we have an error. I don't think that additional check is needed

Reply:
A `CHECK` _is_ a way to ensure that your assumptions are correct.
It behaves better in that you get better stack frames and no memory corruption and random side effects if your assumptions are incorrect.
It's also a way to "document" your assumptions in a better and more concise way than a comment